File: Divide-and-Conquer Searching/Binary Search/Search Insert Position/SearchInsertPosition.c
#include <stdio.h>

int searchInsertPosition(int arr[], int size, int target)
{
    int left = 0;
    int right = size - 1;

    while (left <= right)
    {
        int mid = left + (right - left) / 2;

        if (arr[mid] == target)
        {
            return mid;
        }
        else if (arr[mid] < target)
        {
            left = mid + 1;
        }
        else
        {
            right = mid - 1;
        }
    }

    return left;
}

int main()
{
    int arr[] = {1, 3, 5, 6};
    int size = sizeof(arr) / sizeof(arr[0]);
    int target = 5;

    int position = searchInsertPosition(arr, size, target);

    printf("Position: %d\n", position);

    return 0;
}

File: Divide-and-Conquer Searching/Binary Search/Search Insert Position/SearchInsertPosition.py
def search_insert_position(arr, target):
    left = 0
    right = len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2

        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return left


if __name__ == "__main__":
    arr = [1, 3, 5, 6]
    target = 5

    position = search_insert_position(
        arr,
        target
    )

    print("Position:", position)
